File: builds/web/web/Models/BuildStatus/BuildStatus.cs
namespace web.Models.BuildStatus
{
    using System;

    public class BuildStatus : ItemPropertyStatus
    {
        public BuildTargetState DebugState { get; set; }

        public BuildTargetState ReleaseState { get; set; }

        public DateTime CompletedTime { get; set; }
    }
}

File: builds/app/scripts/app/builds/models/buildStatus/buildStatus.ts
import ItemPropertyStatus = require('../itemPropertyStatus');

'use strict';

class BuildStatus extends ItemPropertyStatus {
    public debugState: number;
    public releaseState: number;
    public completedTime: Date;
    
    constructor(state: number, debugState: number, releaseState: number, completedTime: Date) {
        super(state);
        this.debugState = debugState;
        this.releaseState = releaseState;
        this.completedTime = completedTime;
    }
}

export = BuildStatus;
